paasta_tools/cli/cmds/remote_run.py

```diff
@@ -0,0 +1,314 @@
+            driver.stop() # is this the right place to stop the driver?
+
+
+    def new_task(self, offer):
+        task = self.service_config.base_task(self.system_paasta_config)
+        id = uuid.uuid4()
+        task.task_id.value = str(id)
+        task.slave_id.value = offer.slave_id.value
+
+        if self.command:
+            task.command.value = self.command
+
+        return task
+
+
+def add_subparser(subparsers):
+    list_parser = subparsers.add_parser(
+        'remote-run',
+        help="Schedule Mesos to run adhoc command in context of a service",
+        description=(
+            "'paasta remote-run' is useful for running adhoc commands in context "
+            "of a service's Docker image. The command will be scheduled on a "
+            "Mesos cluster and stdout/stderr printed after execution is finished."
+        ),
+        epilog=(
+            "Note: 'paasta remote-run' Mesos API that may require authentication."
+        ),
+    )
+    list_parser.add_argument(
+        '-s', '--service',
+        help='The name of the service you wish to inspect',
+    ).completer = lazy_choices_completer(list_services)
+    list_parser.add_argument(
+        '-c', '--cluster',
+        help=("The name of the cluster you wish to run your task on. "
+              "If omitted, uses the default cluster defined in the paasta remote-run configs"),
+    ).completer = lazy_choices_completer(list_clusters)
+    list_parser.add_argument(
+        '-y', '--yelpsoa-config-root',
+        dest='yelpsoa_config_root',
+        help='A directory from which yelpsoa-configs should be read from',
+        default=DEFAULT_SOA_DIR,
+    )
+    list_parser.add_argument(
+        '--json-dict',
+        help='When running dry run, output the arguments as a json dict',
+        action='store_true',
+        dest='dry_run_json_dict',
+    )
+    list_parser.add_argument(
+        '-C', '--cmd',
+        help=('Run Docker container with particular command, '
+              'for example: "bash". By default will use the command or args specified by the '
+              'soa-configs or what was specified in the Dockerfile'),
+        required=False,
+        default=None,
+    )
+    list_parser.add_argument(
+        '-i', '--instance',
+        help=("Simulate a docker run for a particular instance of the service, like 'main' or 'canary'"),
+        required=False,
+        default=None,
+    ).completer = lazy_choices_completer(list_instances)
+    list_parser.add_argument(
+        '-v', '--verbose',
+        help='Show Docker commands output',
+        action='store_true',
+        required=False,
+        default=True,
+    )
+    list_parser.add_argument(
+        '-d', '--dry-run',
+        help='Don\'t launch the task',
+        action='store_true',
+        required=False,
+        default=False,
+    )
+    list_parser.add_argument(
+        '-l', '--local',
+        help='Run mesos-related bits locally, rather than ssh\'ing to mesos-master',
```

Comment:
actually i thought having that flag would help with debugging, but also right now the flow is:

    1. paasta remote-run ...
    2. `- ssh mesos-master ...
    3.    `- paasta remote-run ... -l
    4.       `- "curl" mesos-master ...

in absence of `-l` flag how/where is the decision made whether to run ssh (1) or to talk to mesos (2)? would ssh-invoking code would be a separate cli tool or is there some way to reliably say that `$0` is already running on mesos-master?